src/askui/models/shared/base_agent.py

```diff
@@ -238,3 +241,11 @@ def _maybe_filter_to_n_most_recent_images(
                     new_content.append(content)
                 tool_result.content = new_content
         return messages
+
+    def _handle_stop_reason(self, message: MessageParam) -> None:
+        if message.stop_reason == "max_tokens":
+            error_msg = f"Model stopped due to reaching maximum token limit of {self._settings.max_tokens} tokens"  # noqa: E501
+            raise MaxTokensExceededError(self._settings.max_tokens, error_msg)
+        if message.stop_reason == "refusal":
+            error_msg = "Model refused to process the request"
+            raise ModelRefusalError(message=error_msg)
```

Comment:
Why not just go with the default error messages and passing the reason (or removing the `reason` if not necessary)?